test/suite/test_prepare02.py

```diff
@@ -0,0 +1,136 @@
+        with self.expectedStderrPattern(" not permitted in a"):
+            self.assertRaises(wiredtiger.WiredTigerError,
+                lambda: self.session.open_cursor("table:mytable", None))
+        with self.expectedStderrPattern(" not permitted in a"):
+            self.assertRaises(wiredtiger.WiredTigerError,
+                lambda: self.session.alter("table:mytable",
+                   "access_pattern_hint=random"))
+        with self.expectedStderrPattern(" not permitted in a"):
+            self.assertRaises(wiredtiger.WiredTigerError,
+                lambda: self.session.create("table:mytable1",
+                   "key_format=S,value_format=S"))
+        with self.expectedStderrPattern(" not permitted in a"):
+            self.assertRaises(wiredtiger.WiredTigerError,
+                lambda: self.session.rebalance("table:mytable", None))
+        with self.expectedStderrPattern(" not permitted in a"):
+            self.assertRaises(wiredtiger.WiredTigerError,
+                lambda: self.session.rename("table:mytable",
+                   "table:mynewtable", None))
+        with self.expectedStderrPattern(" not permitted in a"):
+            self.assertRaises(wiredtiger.WiredTigerError,
+                lambda: self.session.salvage("table:mytable", None))
+        with self.expectedStderrPattern(" not permitted in a"):
+            self.assertRaises(wiredtiger.WiredTigerError,
+                lambda: self.session.truncate("table:mytable",
+                    None, None, None))
+        with self.expectedStderrPattern(" not permitted in a"):
+            self.assertRaises(wiredtiger.WiredTigerError,
+                lambda: self.session.drop("table:mytable", None))
+        with self.expectedStderrPattern(" not permitted in a"):
+            self.assertRaises(wiredtiger.WiredTigerError,
+                lambda: self.session.join(cursor, cursor,
+                    "compare=gt,count=10"))
+        with self.expectedStderrPattern("logging not enabled"):
+            self.assertRaises(wiredtiger.WiredTigerError,
+                lambda: self.session.log_flush("sync=on"))
+        with self.expectedStderrPattern(" not permitted in a"):
+            self.assertRaises(wiredtiger.WiredTigerError,
+                lambda: self.session.upgrade("table:mytable", None))
+        with self.expectedStderrPattern(" not permitted in a"):
+            self.assertRaises(wiredtiger.WiredTigerError,
+                lambda: self.session.verify("table:mytable", None))
+        with self.expectedStderrPattern(" not permitted in a"):
+            self.assertRaises(wiredtiger.WiredTigerError,
+                lambda: self.session.snapshot("name=test"))
+        with self.expectedStderrPattern(" not permitted in a"):
+            self.assertRaises(wiredtiger.WiredTigerError,
+                lambda: self.session.timestamp_transaction(
+                    "commit_timestamp=2a"))
+        self.session.log_printf("Wrote 100 records")
```

Comment:
It isn't clear to me why `log_printf` and `log_flush` are allowed. There are other, equally non-active-transaction methods that we're forbidding such as `transaction_pinned_range` and some other cursor types, etc. These need comments both in the WT code and test code if they stay.

Some session methods are not listed here, such as `transaction_pinned_range` and `log_flush`. (Related to earlier comment about alphabetizing to make it easier to detect missing methods.)